### PR TITLE
feat: Table support renderFilterDropdown #2015

### PR DESCRIPTION
**Pull request by shijiatongxue**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [x] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [x] Feature

### PR description

Feat #2015

### Changelog
🇨🇳 Chinese
- Feat: Table 支持使用 renderFilterDropdown 自定义筛选器 dropdown 内容 #2015

---

🇺🇸 English
- Fix: Table support renderFilterDropdown #2015

### Checklist
- [x] Test or no need
- [x] Document or no need
- [x] Changelog or no need

### Other
- [ ] Skip Changelog

### CODE REVIEW

File: packages/semi-ui/table/interface.ts [hunk 1]
```diff
@@ -22,6 +21,7 @@ import {
     BaseEllipsis
 } from '@douyinfe/semi-foundation/table/foundation';
 import { ScrollDirection, CSSDirection } from 'react-window';
+import { ColumnFilterProps } from './ColumnFilter';
```
Comment (pointhalo) [inline packages/semi-ui/table/interface.ts:24]: 这里加一个type 显式声明吧 import type {xx} from xx。storybook会把这个当做 js module 去import，会有个warning。ts+pnpm有些特殊case下有时候也会有问题。

Comment (shijiatongxue) [reply packages/semi-ui/table/interface.ts:24]: get

File: packages/semi-ui/table/ColumnFilter.tsx [hunk 2]
```diff
@@ -128,27 +131,72 @@ function renderDropdown(props: RenderDropdownProps = {}, nestedElem: React.React
...
+    const isCustomDropdownVisible = !isFilterDropdownVisibleControlled && isCustomFilterDropdown;
+    const [selectedKeys, setSelectedKeys] = useState<any[]>(filteredValue);
+    const dropdownVisibleInitValue = isCustomDropdownVisible ? false : filterDropdownVisible;
+    const [dropdownVisible, setDropdownVisible] = useState<boolean | undefined>(dropdownVisibleInitValue);
+
+    useEffect(() => {
+        if (typeof filterDropdownVisible !== 'undefined') {
+            setDropdownVisible(filterDropdownVisible);
+        }
+    }, [filterDropdownVisible]);
```
Comment (pointhalo) [inline packages/semi-ui/table/ColumnFilter.tsx:160]: 这个写法，很难 fa诶

File: packages/semi-ui/table/ColumnFilter.tsx [hunk 3]
```diff
@@ -177,32 +228,59 @@ export default function ColumnFilter(props: ColumnFilterProps = {}): React.React
...
         filterDropdown :
-        typeof renderFilterDropdown === 'function' ?
-            renderFilterDropdown(renderProps, { iconElem }) :
-            renderDropdown(renderProps, iconElem);
+        renderDropdown(renderProps, iconElem);
 
     return filterDropdown;
 }
 
-export interface OnSelectData {
```
Comment (pointhalo) [inline packages/semi-ui/table/ColumnFilter.tsx:191]: 这个export 不再保留？是否影响原有使用，有使用场景吗？

**Pull request merged**